build.gradle

```diff
@@ -107,6 +127,31 @@ List<String> resolveIntelliJVersions(String versionKey) {
   return [versionValue]
 }
 
+String checkForNewEap (String intellijMajorVersion) {
+  Document htmlContent = Jsoup.connect(project.intellijSnapshotsUrl).get()
+  Element link = htmlContent.select('a[href^='+project.intellijSnapshotsUrl+'com/jetbrains/intellij/idea/ideaIC/'+intellijMajorVersion+'.][href$="-EAP-SNAPSHOT.pom"]')[0]
+  if (link) {
+    String attr = link.attr("href")
+    Pattern pattern = Pattern.compile("ideaIC-(.*?).pom")
+    Matcher matcher = pattern.matcher(attr)
+    matcher.find()
+    return matcher.group(1)
+  }
+}
+
+task updateEapOfLatestSupportedMajor {
+  Properties properties = intellijVersionsProperties()
+  File propFile = intellijVersionsFile()
+  String latestEap = project.intellijVersions.eapOfLatestSupportedMajor
+  char[] latestStable = project.intellijVersions.latestStable.toCharArray()
+  String newerVersion = latestEap ? project.checkForNewEap(latestEap.split("\\.")[0]) :
+            project.checkForNewEap(Character.toString(latestStable[2]) + Character.toString(latestStable[3]) + Character.toString(latestStable[5]+1))
```

Comment:
Hmmm the problem with `+1` here is that e.g. after `213` (2021.3) the next version was `221` (2022.1), not `214`.

Reply:
As a remedy, I'd rework the logic in `checkForNewEap` to check for a version **equal or higher** to the specified `intellijMajorVersion`

Reply:
Also! note that you've got `intellijVersionToBuildNumber` helper for converting `latestStable` to build number